src/pages/api/twitter-request-token.ts

```diff
@@ -0,0 +1,94 @@
+import { createHmac, randomBytes } from "crypto"
+import { NextApiHandler } from "next"
+
+const CONSUMER_KEY = process.env.TWITTER_API_KEY
+const CONSUMER_SECRET = process.env.TWITTER_API_SECRET
+const SIGNATURE_METHOD = "HMAC-SHA1"
+const OAUTH_VERSION = "1.0"
+const DEFAULT_CALLBACK_URL = "https://guild.xyz/oauth"
+const BASE_URL = "https://api.twitter.com/oauth/request_token"
+const KEY = `${encodeURIComponent(CONSUMER_SECRET)}&`
+const PARSE_FAIL_ERROR_MSG = "Unexpected data received from Twitter"
+
+// /oauth_token=(.*?)&oauth_token_secret=(.*?)&oauth_callback_confirmed=true/
+const parseV1Response = (response: string) => {
```

Comment:
I think we could use `URLSearchParams` here! ;)
Haven't tested this code, but something like this maybe:
```ts
const params = new URLSearchParams(response)
const oauthToken = params.get("oauth_token")
const oauthTokenSecret = params.get("oauth_token_secret")
```

Reply:
Great idea! Tested it out locally, and seems to work fine! Changed in https://github.com/agoraxyz/guild.xyz/pull/852/commits/cdf595693bedaf47b9454f24969cd94e888a69ec
Also made this change in the Twitter connector 👍